prov/sockets/src/sock_av.c

```diff
@@ -386,15 +386,18 @@ static int sock_av_insertsym(struct fid_av *av, const char *node, size_t nodecnt
 		fmt = 0;
 	else 
 		fmt = offset;
-
+	
+	assert((hostlen-offset) < FI_NAME_MAX);
 	strncpy(base_host, node, hostlen - (offset));
 	var_port = atoi(service);
 	var_host = atoi(node + hostlen - offset);
 	
 	for (i = 0; i < nodecnt; i++) {
 		for (j = 0; j < svccnt; j++) {
-			sprintf(tmp_host, "%s%0*d", base_host, fmt, var_host + i);
-			sprintf(tmp_port, "%d", var_port + j);
+			ret = snprintf(tmp_host, FI_NAME_MAX, "%s%0*d", base_host, fmt, var_host + i);
+			assert(ret < FI_NAME_MAX);
+			ret = snprintf(tmp_port, FI_NAME_MAX,  "%d", var_port + j);
+			assert(ret < FI_NAME_MAX);
```

Comment:
I don't think we need to assert that snprintf() works as defined.  Plus it can return FI_NAME_MAX.  If we need to check for errors, we should check for a negative value.